### PR TITLE
fix(trashbin): Correctly set original name as displayname

### LINKED ISSUE #46657: [Bug]: Deleted items shows random number (timestamp) after item name
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

If I delete a folder/file, the name of this looks like:
image

### Steps to reproduce

1. Update to 29.0.4
2. Delete a folder
3. Go to the trashbin

### Expected behavior

no random suffix like .d1721463802

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

29

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.3

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated from a MINOR version (ex. 28.0.1 to 28.0.2)

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: https://github.com/nextcloud/server/issues/46657

## Summary

If you like I can split this (see both commits):
1. Fix trashbin (see linked issue) display name as it should be set to the WebDAV displayname property
2. Improve trashbin view performance by only issue one PROPFIND, this also allows to clean up the code a lot.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files_trashbin/lib/Sabre/TrashbinPlugin.php [hunk 1]
```diff
@@ -74,6 +74,11 @@ public function propFind(PropFind $propFind, INode $node) {
 			return $node->getDeletedBy()?->getDisplayName();
 		});
 
+		// Pass the real filename as the DAV display name
+		$propFind->handle(FilesPlugin::DISPLAYNAME_PROPERTYNAME, function () use ($node) {
+			return $node->getFilename();
+		});
```
Comment (skjnldsv) [inline apps/files_trashbin/lib/Sabre/TrashbinPlugin.php:80]: Maybe deprecate `nc:trashbin-filename` then?

**Review by skjnldsv — APPROVED** [review]

**Review by come-nc — APPROVED** [review]

**Comment by icewind1991** [comment]
This broke navigating into folders: https://github.com/nextcloud/server/pull/47153

File: apps/files_trashbin/lib/Sabre/TrashbinPlugin.php (hunk 1 above)
Comment (skjnldsv) [reply apps/files_trashbin/lib/Sabre/TrashbinPlugin.php:80]: @susnux

Comment (susnux) [reply apps/files_trashbin/lib/Sabre/TrashbinPlugin.php:80]: We can do so, but wanted not to do this here as a bit unrelated. can do so in another PR.

**Pull request merged**